lib/private/DB/Adapter.php

```diff
@@ -31,8 +31,25 @@ public function __construct($conn) {
 	 * @return int id of last insert statement
 	 * @throws Exception
 	 */
-	public function lastInsertId($table) {
-		return (int)$this->conn->realLastInsertId($table);
+	public function lastInsertId($table, bool $allowRetry = true) {
+		$return = $this->conn->realLastInsertId($table);
+		if ($return === false || $return === null) {
+			return (int)$return;
```

Comment:
I'm confused by this. If the first try already fails it just returns, no?

Reply:
At this point `$return` is false|null, which turns to into int(0)

Reply:
Maybe it would be clearer to write directly

```suggestion
		if ($return === false || $return === null) {
			return 0;
```